directed_work/03_deploy_api/README.md

```diff
@@ -0,0 +1,318 @@
+  ✓ Creating Container Repository...                          
+  ✓ Uploading sources...                                     
+  . Building Container...                                    
+  . Creating Revision...                                    
+  . Routing traffic...                                
+  . Setting IAM Policy...                              
+Deployment failed                                      
+ERROR: (gcloud.run.deploy) PERMISSION_DENIED: The caller does not have permission
+```
+
+This error may be due to the fact that you don't have the right permissions to deploy the app. Make sure you're logged in with the right account and that you have the right permissions to deploy the app.
+
+```
+> gcloud auth login
+```
+
+## 4. Monitor the API logs and usage in Google Cloud
+
+1. In the GCP console, search for `Cloud Run` and click on the first result. You should see the following screen:
+![Cloud Run](img/3.png)
+2. Click on the service you just created. You should see the different metrics of your service and above the URL of your service.
+3. Click on `Logs` (or `Journaux` if you are using the French version of the console) to view the logs of your service.
+![Service](img/4.png) 
+
+
+## 5. Clean up
+
+1. In `Cloud Run` tab, select the service you just created and click on `Delete` to delete the service
+![Delete](img/5.png)
+4. In the GCP console, search for the `Artifact Registry` and click on it.
+5. Select the `cloud-run-source-deploy` repository and `Delete` to delete the repository
+![Delete](img/6.png)
+
+
+## 6. About pricing
+
+Here we built a small app and deployed it to GCP. We used the `Cloud Run` service which is a serverless platform that allows you to run stateless containers that are invocable via HTTP requests. The pricing of `Cloud Run` is based on the number of requests and the amount of memory used by the app. You can find more information about the pricing of `Cloud Run` [here](https://cloud.google.com/run/pricing). But to give you an idea, in terms of processing power, the first 240k vCPU-seconds per months are free. A vCPU-second is a unit of time that represents the processing power of a single virtual CPU running for one second. After the first 240k vCPU-seconds, you will be charged 0,00001800\$ per vCPU-seconde. So for example if you have an app that uses 1 vCPU for 1 second, you will be charged 0,00001800\$ after the first 240k vCPU-seconde. This [website](https://phoenixnap.com/kb/what-is-a-vcpu) explains what is a vCPU and how you can calculate vCPU requirements for your app..
+
+In terms of memory, the first 450k Gio-seconde per month are free. After that, you will be charged 0,00000200\$ per Gio-seconde. A Gio-seconde is a unit of time that represents the amount of memory used by the app for one second. So for example if you have an app that uses 1 Gio of memory for 1 second, you will be charged 0,00000200\$ after the first 450k Gio-seconde.
```

Comment:
Add a part on "scale to zero": https://cloud.google.com/run/docs/about-instance-autoscaling